File: Data Structures & Algorithms/is-subsequence/submission-0.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) == 0:
            return True
        if len(s) > len(t):
            return False
        i = 0
        for j in t:
            if len(s) == i:
                return True
            if s[i] == j:
                i += 1
        return len(s) == i
        

File: Data Structures & Algorithms/combinations/submission-0.cpp
class Solution {
public:
    void combinations(vector<vector<int>>& ans, vector<int>& com, int n, int k,
                      int idx) {
        if (com.size() == k) {
            ans.push_back(com);
            return;
        }
        for (int i = idx; i <= n; i++) {
            com.push_back(i);
            combinations(ans, com, n, k, i + 1);
            com.pop_back();
        }
    }
    vector<vector<int>> combine(int n, int k) {
        vector<vector<int>> ans;
        vector<int> com;
        combinations(ans, com, n, k, 1);

        return ans;
    }
};